account_payment_order_notification/wizard/wizard_account_payment_order_notification.py

```diff
@@ -59,7 +63,8 @@ def action_process(self):
         notifications = []
         for item in self.line_ids.filtered("to_send"):
             payment_line_ids = self.order_id.payment_line_ids.filtered(
-                lambda x: x.partner_id == item.partner_id
+                lambda x: (x.move_line_id.move_id.partner_id or x.partner_id)
+                == item.partner_id
             )
             data = {
                 "partner_id": item.partner_id.id,
```

Comment:
This one should change as well, isn't it? Can't you avoid the `or` each time?

Reply:
IMO previously is correct, although a bit confusing, it is changed.

Reply:
`line_ids` records already have the correct `partner_id`.

Reply:
OK, got it.